backend/app/rest/api/rest_private.go

```diff
@@ -314,6 +317,55 @@ func (s *private) sendEmailConfirmationCtrl(w http.ResponseWriter, r *http.Reque
 	render.JSON(w, r, R.JSON{"user": user, "address": address})
 }
 
+// sendTelegramConfirmationCtrl gets address and siteID from query, makes confirmation token and sends it to user.
+// GET /telegram/subscribe?site=siteID&address=@someone
+//nolint:dupl // too hard to deduplicate that logic, as then it's tricky to use SendErrorJSON
+func (s *private) sendTelegramConfirmationCtrl(w http.ResponseWriter, r *http.Request) {
+	user := rest.MustGetUserInfo(r)
+	address := r.URL.Query().Get("address")
+	siteID := r.URL.Query().Get("site")
+	if address == "" {
+		rest.SendErrorJSON(w, r, http.StatusBadRequest,
+			errors.New("missing parameter"), "address parameter is required", rest.ErrInternal)
+		return
+	}
+	existingAddress, err := s.dataService.GetUserTelegram(siteID, user.ID)
+	if err != nil {
+		log.Printf("[WARN] can't read telegram for %s, %v", user.ID, err)
```

Comment:
not following the loginc here. is it even make sense to continue afte  failed  GetUserTelegram? If it is, we need some comment/explanation

Reply:
Addressed in the second commit.